caikit/core/data_model/base.py

```diff
@@ -618,9 +618,12 @@ def from_proto(cls, proto):
 
             elif field in cls._fields_message:
                 if proto.HasField(field):
-                    contained_class = cls.get_class_for_proto(proto_attr)
-                    contained_obj = contained_class.from_proto(proto_attr)
-                    kwargs[field] = contained_obj
+                    if proto_attr.DESCRIPTOR.full_name == "google.protobuf.Struct":
+                        kwargs[field] = json_dict.struct_to_dict(proto_attr)
+                    else:
+                        contained_class = cls.get_class_for_proto(proto_attr)
+                        contained_obj = contained_class.from_proto(proto_attr)
```

Comment:
Is it possible to avoid this `if` case (and the related one below) by having our own `JsonDict` data model class that has a `.from_proto` and `.fill_proto`?

Reply:
Yeah, I thought about that. The blocker I hit trying to go down this route is that we want the authors of the block to be able to treat the object as a plain-old `dict`. I tried to create an `@dataobject` that inherits from both `DataObjectBase` and `dict`, but it blew up spectacularly. Rather than go down the route of making a `JsonDict` object "quack like a dict," I figured this was cleaner.

Reply:
Yeah... I thought that 
> making a JsonDict object "quack like a dict,"

would be pretty simple, but it's probably not lol. As is I think we can also swap out the definition of a `JsonDict` to a data model class that acts like a dict pretty safely later if we want